torch/_functorch/partitioners.py

```diff
@@ -808,8 +813,7 @@ def should_ban_recomputation(node):
             return False
         if node.target in [aten.lift_fresh_copy.default, aten.lift_fresh.default]:
             return False
-        # NB: "recompute" == 0 means that must save this node.
-        if node.meta.get("recompute", None) == 0:
+        if node.meta.get("recompute", None) == CheckpointPolicy.MUST_SAVE:
```

Comment:
It looks like right above this condition, if the op is a view or a `lift_fresh_copy` then we will not ban recomputation, even if the user marked as MUST_SAVE.

Do you think we should either error here, or prefer the user annotation instead?

Reply:
This is a good point, I think we should error when user MUST_SAVE a view or lift_fresh, etc. rather than ignore the annotation.

However, generally it is tricky to absolutely respect the "MUST_SAVE" condition, e.g. in the case where some code get's DCE'd. 

Or another case is if the user choose to MUST_SAVE some tensor, but let's say the backward formula reduces this value before using it, so maybe we'd rather save the value post reduction.

But these two are really just the consequence of the fact that MUST_NOT_RECOMPUTE became MUST_SAVE... maybe its worth renaming it back lol.